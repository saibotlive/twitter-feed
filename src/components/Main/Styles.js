import styled, { injectGlobal } from 'styled-components';

injectGlobal`
* {
box-sizing: border-box;
}
body {
  font-size: 16px;
  font-family: Helvetica, Arial, sans-serif;
  background: lightgrey;
}
`;

export const Wrapper = styled.div`
  position: relative;
  width: 100%;
  background-image: url(${props => props.bgURL});
  background-size: cover;
`;

export const Container = styled.div`
  position: relative;
  max-width: 1024px;
  width: 100%;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: space-between;
  padding: 20px;
`;

export const ErrorMessage = styled.h1`
  text-align: center;
`;
