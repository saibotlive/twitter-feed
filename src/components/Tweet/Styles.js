import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
  width: 100%;
  padding: 20px;
  margin: 0 0 20px 0;
  border-radius: 20px;
  background: white;

  @media screen and (min-width: 600px) {
    width: 48%;
  }
`;

export const Header = styled.div`
  font-size: 1.5rem;
  margin: 0 0 20px 0;
`;

export const Link = styled.a`
  font-size: 1.5rem;
  text-decoration: none;
  color: ${props => `#${props.color}`};
  word-break: break-word;
  display: inline-block;
  margin: 0 0 20px 0;
`;

export const Img = styled.img`
  display: block;
  max-width: 100%;
  border-radius: 5px;
  margin: 0 auto;
`;
