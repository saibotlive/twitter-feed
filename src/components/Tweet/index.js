import React from 'react';
import PropTypes from 'prop-types';
import { Container, Header, Link } from './Styles';

const Tweet = ({ tweet }) => {
  const { entities, text, user } = tweet;
  const withLink = text.indexOf('http') !== -1;
  const links = text.slice(text.indexOf('http')).split(' ');

  const headerTxt = withLink ? text.slice(0, text.indexOf('http')) : text;

  return (
    <Container>
      <Header>{headerTxt}</Header>
      {withLink &&
        entities.urls.map((link, i) => (
          <Link
            color={user.profile_link_color}
            href={link.url}
            key={`link-${i}`}
            target="_blank"
          >
            {links[i]}
          </Link>
        ))}
    </Container>
  );
};

Tweet.propTypes = {
  tweet: PropTypes.shape({
    entities: PropTypes.object.isRequired,
    text: PropTypes.string.isRequired,
    user: PropTypes.object.isRequired
  })
};

export default Tweet;
