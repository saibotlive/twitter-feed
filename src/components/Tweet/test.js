import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import { Header, Link, Img } from './Styles';
import Tweet from './';

const props = {
  tweet: {
    entities: {
      urls: [{ display_url: 'display', expanded_url: 'expanded' }]
    },
    text: 'This is a tweet http://twitter.com',
    user: {
      profile_background_image_url: 'url',
      profile_link_color: '000000'
    }
  }
};

it('renders Tweet component', () => {
  const comp = shallow(<Tweet {...props} />);
  expect(comp).toMatchSnapshot();
});

it('renders Tweet text', () => {
  const comp = shallow(<Tweet {...props} />);
  expect(comp.find(Header).prop('children')).toBe('This is a tweet ');
});

it('renders Tweet without Link', () => {
  const newProps = { ...props, tweet: { ...props.tweet, text: 'No link' } };
  const comp = shallow(<Tweet {...newProps} />);
  expect(comp.find(Link).exists()).toBe(false);
});

it('renders Tweet with Img', () => {
  const newProps = {
    ...props,
    tweet: {
      ...props.tweet,
      entities: { ...props.tweet.entities, media: [{ media_url: 'img-url', id_str: 'str' }] }
    }
  };
  const comp = shallow(<Tweet {...newProps} />);
  console.log('xtest', comp.html());
  expect(comp.find(Img).exists()).toBe(true);
});

it('renders Tweet Link with hex color black', () => {
  const comp = shallow(<Tweet {...props} />);
  expect(comp.find(Link)).toHaveStyleRule('color', '#000000');
});
