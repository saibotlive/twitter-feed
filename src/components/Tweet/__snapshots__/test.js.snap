// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tweet component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tweet
    tweet={
        Object {
            "entities": Object {
              "urls": Array [
                Object {
                  "display_url": "display",
                  "expanded_url": "expanded",
                },
              ],
            },
            "text": "This is a tweet http://twitter.com",
            "user": Object {
              "profile_background_image_url": "url",
              "profile_link_color": "000000",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          This is a tweet 
</styled.div>,
        Array [
          <styled.a
            color="000000"
            href={undefined}
            target="_blank"
>
            http://twitter.com
</styled.a>,
        ],
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is a tweet ",
        },
        "ref": null,
        "rendered": "This is a tweet ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "link-0",
        "nodeType": "class",
        "props": Object {
          "children": "http://twitter.com",
          "color": "000000",
          "href": undefined,
          "target": "_blank",
        },
        "ref": null,
        "rendered": "http://twitter.com",
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            This is a tweet 
</styled.div>,
          Array [
            <styled.a
              color="000000"
              href={undefined}
              target="_blank"
>
              http://twitter.com
</styled.a>,
          ],
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is a tweet ",
          },
          "ref": null,
          "rendered": "This is a tweet ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "link-0",
          "nodeType": "class",
          "props": Object {
            "children": "http://twitter.com",
            "color": "000000",
            "href": undefined,
            "target": "_blank",
          },
          "ref": null,
          "rendered": "http://twitter.com",
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
